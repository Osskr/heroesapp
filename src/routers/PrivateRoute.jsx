import React from 'react'
import { Redirect, Route } from 'react-router-dom'
import {PropTypes} from 'prop-types'
export const PrivateRoute = ({isAuthenticated,component: Component, ...rest}) => {
    
    
    console.log(rest.location.pathname )
    //guardamos la ruta en el local storage
    localStorage.setItem('lastPath',rest.location.pathname)
    return (
        <Route {...rest}  
               component={ (props)=>(
                          (isAuthenticated)? (<Component {...props}/>)
                          : (<Redirect to='login'/>)
                )}/>

     
    )
}
 
PrivateRoute.propTypes ={

    isAuthenticated : PropTypes.bool.isRequired,
    component: PropTypes.func.isRequired,
       
}

