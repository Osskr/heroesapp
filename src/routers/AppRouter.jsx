import React, { useContext } from "react";
import {
  BrowserRouter as Router,
  Switch,
  }
from "react-router-dom";
import { AuthContext } from "../auth/AuthContex";

import { LoginScreen } from "../components/login/LoginScreen";
import { DashBoardRoutes } from "./DashBoardRoutes";
import { PrivateRoute } from "./PrivateRoute";
import { PublicRoute } from "./PublicRoute";

export const AppRouter = () => {

 const {user} = useContext(AuthContext)
  
    return (
        <div>
            <Router>
                <div>                                    
                    <Switch>
                        <PublicRoute  isAuthenticated={user.logged} exact path="/login" component={LoginScreen}/>
                        <PrivateRoute path="/"
                                      component={DashBoardRoutes}
                                      isAuthenticated={user.logged}/>
                    </Switch>                
                </div>
            </Router>
        </div>
    )
}
