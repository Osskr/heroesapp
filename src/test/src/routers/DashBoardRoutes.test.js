import React from 'react'
import {mount} from 'enzyme'
import { DashBoardRoutes } from '../../../routers/DashBoardRoutes'
import { MemoryRouter } from 'react-router-dom'
import { AuthContext } from '../../../auth/AuthContex'


describe('Pruebas sobre el DashBoardRouter', () => {
  test('should mostrarse correctamente', () => {
    
    const contextValue = {
      dispatch:jest.fn(),
      user:{
        logged:true,
        name: 'atr-perri'
      }
    }
    const wrapper = mount(<MemoryRouter>
                            <AuthContext.Provider value={contextValue}>
                             <DashBoardRoutes/>
                             </AuthContext.Provider>
                          </MemoryRouter>)
                                 
  
    expect(wrapper).toMatchSnapshot()
    
    expect(wrapper.find('.user-name').text().trim()).toBe('atr-perri')
  })



  
  
})
