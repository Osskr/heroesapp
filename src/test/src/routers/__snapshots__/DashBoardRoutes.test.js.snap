// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el DashBoardRouter should mostrarse correctamente 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Heroes
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link user-name"
        >
          atr-perri
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>
  <div
    className="container mt-2"
  >
    <div>
      <h1>
        Marvel Heroes
      </h1>
      <hr />
      <div
        className="card-columns animate__animated animate__fadeIn"
      >
        <div
          className="card"
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          <img
            alt="..."
            className="card-img-top"
            src="./assets/heroes/marvel-spider.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="mb-0"
            >
              Spider Man
            </h5>
            <p
              className="card-text my-0"
            >
              Peter Parker
            </p>
            <p
              className="card-text "
            >
              <small>
                Amazing Fantasy #15
              </small>
            </p>
            <span
              className="badge badge-dark font-weight-bold"
            >
              Marvel Comics
            </span>
            <p>
              <a
                href="/hero/marvel-spider"
                onClick={[Function]}
              >
                Mas info
              </a>
            </p>
          </div>
        </div>
        <div
          className="card"
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          <img
            alt="..."
            className="card-img-top"
            src="./assets/heroes/marvel-captain.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="mb-0"
            >
              Captain America
            </h5>
            <p
              className="card-text my-0"
            >
              Steve Rogers
            </p>
            <p
              className="card-text "
            >
              <small>
                Captain America Comics #1
              </small>
            </p>
            <span
              className="badge badge-dark font-weight-bold"
            >
              Marvel Comics
            </span>
            <p>
              <a
                href="/hero/marvel-captain"
                onClick={[Function]}
              >
                Mas info
              </a>
            </p>
          </div>
        </div>
        <div
          className="card"
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          <img
            alt="..."
            className="card-img-top"
            src="./assets/heroes/marvel-iron.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="mb-0"
            >
              Iron Man
            </h5>
            <p
              className="card-text my-0"
            >
              Tony Stark
            </p>
            <p
              className="card-text "
            >
              <small>
                Tales of Suspense #39
              </small>
            </p>
            <span
              className="badge badge-dark font-weight-bold"
            >
              Marvel Comics
            </span>
            <p>
              <a
                href="/hero/marvel-iron"
                onClick={[Function]}
              >
                Mas info
              </a>
            </p>
          </div>
        </div>
        <div
          className="card"
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          <img
            alt="..."
            className="card-img-top"
            src="./assets/heroes/marvel-thor.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="mb-0"
            >
              Thor
            </h5>
            <p
              className="card-text my-0"
            >
              Thor Odinson
            </p>
            <p
              className="card-text "
            >
              <small>
                Journey into Myster #83
              </small>
            </p>
            <span
              className="badge badge-dark font-weight-bold"
            >
              Marvel Comics
            </span>
            <p>
              <a
                href="/hero/marvel-thor"
                onClick={[Function]}
              >
                Mas info
              </a>
            </p>
          </div>
        </div>
        <div
          className="card"
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          <img
            alt="..."
            className="card-img-top"
            src="./assets/heroes/marvel-hulk.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="mb-0"
            >
              Hulk
            </h5>
            <p
              className="card-text my-0"
            >
              Bruce Banner
            </p>
            <p
              className="card-text "
            >
              <small>
                The Incredible Hulk #1
              </small>
            </p>
            <span
              className="badge badge-dark font-weight-bold"
            >
              Marvel Comics
            </span>
            <p>
              <a
                href="/hero/marvel-hulk"
                onClick={[Function]}
              >
                Mas info
              </a>
            </p>
          </div>
        </div>
        <div
          className="card"
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          <img
            alt="..."
            className="card-img-top"
            src="./assets/heroes/marvel-wolverine.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="mb-0"
            >
              Wolverine
            </h5>
            <p
              className="card-text my-0"
            >
              James Howlett
            </p>
            <p
              className="card-text "
            >
              <small>
                The Incredible Hulk #180
              </small>
            </p>
            <span
              className="badge badge-dark font-weight-bold"
            >
              Marvel Comics
            </span>
            <p>
              <a
                href="/hero/marvel-wolverine"
                onClick={[Function]}
              >
                Mas info
              </a>
            </p>
          </div>
        </div>
        <div
          className="card"
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          <img
            alt="..."
            className="card-img-top"
            src="./assets/heroes/marvel-daredevil.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="mb-0"
            >
              Daredevil
            </h5>
            <p
              className="card-text my-0"
            >
              Matthew Michael Murdock
            </p>
            <p
              className="card-text "
            >
              <small>
                Daredevil #1
              </small>
            </p>
            <span
              className="badge badge-dark font-weight-bold"
            >
              Marvel Comics
            </span>
            <p>
              <a
                href="/hero/marvel-daredevil"
                onClick={[Function]}
              >
                Mas info
              </a>
            </p>
          </div>
        </div>
        <div
          className="card"
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          <img
            alt="..."
            className="card-img-top"
            src="./assets/heroes/marvel-hawkeye.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="mb-0"
            >
              Hawkeye
            </h5>
            <p
              className="card-text my-0"
            >
              Clinton Francis Barton
            </p>
            <p
              className="card-text "
            >
              <small>
                Tales of Suspense #57
              </small>
            </p>
            <span
              className="badge badge-dark font-weight-bold"
            >
              Marvel Comics
            </span>
            <p>
              <a
                href="/hero/marvel-hawkeye"
                onClick={[Function]}
              >
                Mas info
              </a>
            </p>
          </div>
        </div>
        <div
          className="card"
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          <img
            alt="..."
            className="card-img-top"
            src="./assets/heroes/marvel-cyclops.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="mb-0"
            >
              Cyclops
            </h5>
            <p
              className="card-text my-0"
            >
              Scott Summers
            </p>
            <p
              className="card-text "
            >
              <small>
                X-Men #1
              </small>
            </p>
            <span
              className="badge badge-dark font-weight-bold"
            >
              Marvel Comics
            </span>
            <p>
              <a
                href="/hero/marvel-cyclops"
                onClick={[Function]}
              >
                Mas info
              </a>
            </p>
          </div>
        </div>
        <div
          className="card"
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          <img
            alt="..."
            className="card-img-top"
            src="./assets/heroes/marvel-silver.jpg"
          />
          <div
            className="card-body"
          >
            <h5
              className="mb-0"
            >
              Silver Surfer
            </h5>
            <p
              className="card-text my-0"
            >
              Norrin Radd
            </p>
            <p
              className="card-text "
            >
              <small>
                The Fantastic Four #48
              </small>
            </p>
            <span
              className="badge badge-dark font-weight-bold"
            >
              Marvel Comics
            </span>
            <p>
              <a
                href="/hero/marvel-silver"
                onClick={[Function]}
              >
                Mas info
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
